dotCMS/src/main/java/com/dotmarketing/util/ExportStarterUtil.java

```diff
@@ -0,0 +1,305 @@
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.contenttype.util.ContentTypeImportExportUtil;
+import com.dotcms.publishing.BundlerUtil;
+import com.dotcms.util.transform.TransformerLocator;
+import com.dotmarketing.beans.Clickstream;
+import com.dotmarketing.beans.Clickstream404;
+import com.dotmarketing.beans.ClickstreamRequest;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.beans.Inode;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.beans.PermissionReference;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.cms.factories.PublicCompanyFactory;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.calendar.model.CalendarReminder;
+import com.dotmarketing.portlets.cmsmaintenance.util.AssetFileNameFilter;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.rules.util.RulesImportExportUtil;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.portlets.workflows.model.WorkflowHistory;
+import com.dotmarketing.portlets.workflows.util.WorkflowImportExportUtil;
+import com.dotmarketing.tag.model.Tag;
+import com.dotmarketing.tag.model.TagInode;
+import com.google.common.collect.ImmutableList;
+import com.liferay.portal.ejb.ImageLocalManagerUtil;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+
+public class ExportStarterUtil {
+
+    
+
+    final File outputDirectory;
+    
+    public ExportStarterUtil() {
+        
+        outputDirectory = new File(ConfigUtils.getBackupPath() + File.separator + "backup_" + System.currentTimeMillis());
+        outputDirectory.mkdirs();
+       
+        
+    }
+    
+    public File createStarterWithAssets() throws FileNotFoundException, IOException {
+        
+        moveAssetsToBackupDir();
+        return createStarterData() ;
+        
+    }
+    
+    public File createStarterData() {
+        createXMLFiles() ;
+        
+        return outputDirectory;
+        
+        
+    }
+    
+    
+    /**
+     * This method will pull a list of all tables /classed being managed by
+     * hibernate and export them, one class per file to the backupTempFilePath
+     * as valid XML. It uses XStream to write the xml out to the files.
+     *
+     * @throws ServletException
+     * @throws IOException
+     * @author Will
+     * @throws DotDataException
+     */
+    @CloseDBIfOpened
+    private void createXMLFiles()  {
+
+        Logger.info(this, "Starting createXMLFiles into " + outputDirectory);
+
+        Set<Class> _tablesToDump = new HashSet<Class>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable '_tablesToDump' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)